traitcuration/traits/views.py

```diff
@@ -50,6 +54,34 @@ def update_mapping(request, pk):
     return HttpResponse(json.dumps(model_to_dict(mapping)), content_type="application/json")
 
 
+@login_required
```

Comment:
When I try to do those actions without logging in, I get redirected to a strangely (un)formatted page:

![image](https://user-images.githubusercontent.com/10669118/87939134-7e84fc00-ca8f-11ea-9dd0-651b6947a014.png)

Reply:
It even shows me as being successfully signed in, and when I click on Sign in above, it doesn't ask me for my auth information again. So it just looks like some automatic redirect which is supposed to work doesn't.

As one solution to this problem, we could simply disable all of the buttons when logged out.

Reply:
> This doesn't seem to happen in the deployment in #62, although it exhibits another problem with external datasources. I'm not sure whether the problems are related.

That is because the Celery background worker isn't running in this deployment. You can verify it by checking this panel in the app's main Heroku page 
![image](https://user-images.githubusercontent.com/36301625/87949625-f1519f80-caae-11ea-9d02-4c51755db02e.png)

I have ran out of free Redis instances to deploy so one thing I could do is have the two open PRs share a Redis database, and manually choose which worker we want to run, based on what app you want to review

Reply:
I will remove the decorator and perform the check manually

Reply:
I see that now there are no editing buttons when a user is logged out, so the problem is solved